File: csharp/pangram/Pangram.cs
using System.Linq;

public static class Pangram
{
    public static bool IsPangram(string input)
    {
        return Enumerable.Range('a', 26)
        .Select(x => (char)x)
        .ToArray()
        .Count(c => input.ToLower().Contains(c)) == 26;
    }
}

File: solutions/haskell/pangram/4/src/Pangram.hs
module Pangram (isPangram) where
import Data.Char as Char ( toLower )
import Data.Function ((&))

alphabet :: [Char]
alphabet = ['a' .. 'z']

isPangram :: String -> Bool
isPangram text = 
    let textLength = length alphabet
      in text
        & map toLower
        & contains
        & length
        & (==) textLength

contains :: String -> String
contains text = [c | c <- alphabet, c `elem` text]